lib/cpp/src/thrift/transport/THeaderTransport.cpp

```diff
@@ -394,15 +411,19 @@ void THeaderTransport::setHeader(const string& key, const string& value) {
   writeHeaders_[key] = value;
 }
 
-size_t THeaderTransport::getMaxWriteHeadersSize() const {
+uint32_t THeaderTransport::getMaxWriteHeadersSize() const {
   size_t maxWriteHeadersSize = 0;
   THeaderTransport::StringToStringMap::const_iterator it;
   for (it = writeHeaders_.begin(); it != writeHeaders_.end(); ++it) {
     // add sizes of key and value to maxWriteHeadersSize
     // 2 varints32 + the strings themselves
     maxWriteHeadersSize += 5 + 5 + (it->first).length() + (it->second).length();
   }
-  return maxWriteHeadersSize;
+  if (maxWriteHeadersSize > std::numeric_limits<uint32_t>().max()) {
```

Comment:
I've just realized - if somebody will compile this code on 32bit machine with gcc he will get "comparison always true" warning... just a thought